subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -3866,9 +3858,9 @@ static void enc_req_reused_send(struct ll_conn *conn, struct node_tx **tx)
 		     sizeof(pdu_ctrl_tx->llctrl.enc_req.skdm)));
 
 	/* NOTE: if not sufficient random numbers, ignore waiting */
-	entropy_get_entropy_isr(entropy, pdu_ctrl_tx->llctrl.enc_req.skdm,
-				sizeof(pdu_ctrl_tx->llctrl.enc_req.skdm) +
-				sizeof(pdu_ctrl_tx->llctrl.enc_req.ivm), 0);
+	lll_entropy_get(pdu_ctrl_tx->llctrl.enc_req.skdm,
```

Comment:
lll_entropy_get was introduced to have a light-weight function to randomize the timing of advertising and not meant to provide cyptographically secure randomness. This change seems to lose this intended differentiation. I'm fine with improving the API in general, but I think we should keep a function for "cheap" randomness to be used to randomize the advertising timing and provide another one for cryptographically secure randomness.

Reply:
Is 'entropy' the right word to use then? I think we should use prng an trng so that it becomes easier to know which one is right in that case.

The prng could be implemented like this:
```
/* RAND_MAX assumed to be 32767 */
           int myrand(void) {
               next = next * 1103515245 + 12345;
               return((unsigned)(next/65536) % 32768);
           }

           void mysrand(unsigned int seed) {
               next = seed;
           }
```
Taken from `man 3 rand`

Reply:
Why not use xoroshiro RNG throughout LL code?

Reply:
> Why not use xoroshiro RNG throughout LL code?

I'd rather have the LL code call OS primitives, rather than rely on a specific SW implementation

Reply:
@George-Stefan @alexandru-porosanu-nxp 
If I introduce lll_prng_get, lll_trng_get, and lll_trng_isr_get, and lll_prng_get returns 0 for openisa then it should behave exactly as it is currently on master.

You can then change it however you want for your LLL implementation.

This should also satisfy @wopu-ot who said they used the current lll_entropy_get as non-security related randomness.

Reply:
I agree that the current naming is not great. The suggestion of introducing lll_prng_get, lll_trng_get, and lll_trng_isr_get should work from my point of view. However, "prng" could be the same as "trng" on some platforms as long as it's fast. Maybe lll_fast_rand_get instead of lll_prng_get of would express the intent better?

Reply:
Updated. Added lll_rng_get for the one that can be either prng or trng.